estate/models/estate_property.py

```diff
@@ -0,0 +1,76 @@
+from odoo import api, fields, models, exceptions
+
+
+class Property(models.Model):
+    _name = "estate.property"
+    _description = "Estate properties"
+
+    name = fields.Char("Title", required=True, )
+    description = fields.Text("Description")
+    postcode = fields.Char("Postcode")
+    date_availability = fields.Date("Available From", copy=False, default=fields.Date.add(fields.Date.today(), months=3))
+    expected_price = fields.Float("Expected Price", required=True)
+    selling_price = fields.Float("Selling Price", readonly=True, copy=False)
+    bedrooms = fields.Integer("Bedrooms", default=2)
+    living_area = fields.Integer("Living Area (sqm)")
+    facades = fields.Integer("Facades")
+    garage = fields.Boolean("Garage")
+    garden = fields.Boolean("Garden")
+    garden_area = fields.Integer("Garden Area (sqm)")
+    garden_orientation = fields.Selection([
+        ("north", "North"),
+        ("south", "South"),
+        ("east", "East"),
+        ("west", "West"),
+        ], "Garden Orientation")
+    status = fields.Selection([
+        ("new", "New"),
+        ("offer_received", "Offer Received"),
+        ("offer_accepted", "Offer Accepted"),
+        ("sold", "Sold"),
+        ("cancelled", "Cancelled"),
+        ], default="new", string="Status", required=True, copy=False)
+    active = fields.Boolean(default=True)
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    salesperson_id = fields.Many2one("res.users", string="Salesperson", default=lambda self: self.env.user)
+    buyer_id = fields.Many2one("res.partner", string="Buyer", copy=False)
+    tags_ids = fields.Many2many("estate.property.tag", string="Tags")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+    total_area = fields.Integer("Total Area (sqm)", compute="_compute_total_area")
+    best_offer = fields.Float("Best Offer", compute="_compute_best_offer")
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids")
+    def _compute_best_offer(self):
+        for record in self:
+            if record.offer_ids:
+                record.best_offer = max(record.offer_ids.mapped("price"))
+            else:
+                record.best_offer = 0
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = "north"
+        else:
+            self.garden_area = 0
+            self.garden_orientation = None
+
+    def cancel_property(self):
+        for record in self:
+            if record.status == "sold":
+                raise exceptions.UserError("Sold properties cannot be cancelled")
+            record.status = "cancelled"
```

Comment:
```suggestion
        if self.filtered(lambda property: property.status == "sold"):
            raise exceptions.UserError("Sold properties cannot be cancelled")
        self.status = "cancelled"
```
here you don't even need to loop on the records manually :smile: 
same comment applies for function below